validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorLoader.java

```diff
@@ -33,12 +36,11 @@
   static Map<BLSPublicKey, ValidatorInfo> initializeValidators(ArtemisConfiguration config) {
     // Get validator connection info and create a new ValidatorInfo object and put it into the
     // Validators map
+
     final Map<BLSPublicKey, ValidatorInfo> validators =
-        loadValidatorKeys(config).stream()
-            .collect(
-                Collectors.toMap(
-                    BLSKeyPair::getPublicKey,
-                    blsKeyPair -> new ValidatorInfo(new LocalMessageSignerService(blsKeyPair))));
+        config.getValidatorExternalSigningPublicKeys().isEmpty()
+            ? createLocalSignerValidatorInfo(config)
+            : createExternalSignerValidatorInfo(config);
```

Comment:
It seems potentially confusing that we toggle between the two modes. We could just load both local and external validator info and merge the two which would be more predictable.  Not likely to be a real need of users, just removes one potential gotcha.  I'd prefer the external signer over the local one should a public key appear in both (the external may have extra slashing protection).

Reply:
done